### DISCUSSION TITLE
Allow projects to use an extension other than `.ts` or `.js` for the entrypoint module.

### DISCUSSION THREAD

**Discussion by shirakaba** [NativeScript/rfcs#5] [2. Archives]
NativeScript currently looks for an `{entry}.ts` or `{entry}.js` file as its entrypoint and automatically loads it when the app starts, where `entry` is defined by the `main` field in `package.json`.

**Proposal**: Allow any source file extensions supported by the Webpack configuration to be resolved; not just strictly `ts` and `js`.

**Prior work**: I once opened a PR towards this (https://github.com/NativeScript/nativescript-dev-webpack/pull/875), but the codebase has been refactored since then. It's still largely the same setup, though: https://github.com/NativeScript/nativescript-dev-webpack/blob/5e4936fc6ba9d6503c8fff3e4dd59e541f49d941/index.js#L76-L96

**Benefit**: This would allow React apps to use `app.tsx` or Svelte Native apps to use `app.svelte` as an entrypoint, allowing them to write TSX or Svelte syntax, etc. It's the only reason that React NativeScript has to use this bewildering `React.createElement()` call instead of TSX syntax in the entrypoint:

https://github.com/shirakaba/react-nativescript/blob/8327e29f3b6cd62446ba3ea9b84b3293983b07a0/sample/app/app.ts#L33

**Comment by rigor789** [comment]
I completely agree on this one - in the new webpack implementation (from RFC #2) the `main` field will be used as-is, allowing you to specify any file as the entry point. Another change from the RFC is that the path is now relative to the `package.json` and not the `app` folder. 

I added a note in the rfc:
> BREAKING CHANGES:
>
> - package.json main should now use a relative path to the package.json instead of the app directory
>
>   For example (given we have a src directory where our app is):
>
>   "main": "app.js" becomes "main": "src/app.js"
>
>    This simplifies things, and will allow ctrl/cmd + clicking on the filename in some editors.

**Comment by shirakaba** [comment]
> Another change from the RFC is that the path is now relative to the package.json and not the app folder.

Ah yes, I support changing `"main": "app.js"` to `"main": "src/app.js"`, as it reduces the magic. I brought it up as bewildering in a TSC or community call once. Is there a standard procedure for indicating support for an RFC?

**Comment by rigor789** [comment]
Issues are prior-discussions, but for rfc's you can leave a 👍 on the PR https://github.com/NativeScript/rfcs/pull/2 or comment etc. 

I feel like generally unless there is major opposition to an RFC - and the TSC also approves it (by leaving an approval on the PR), it can proceed to the "active" stage:
https://github.com/NativeScript/rfcs#reviewing-rfcs

**Comment by rigor789** [comment]
I'm closing this - as this is already working/included in the webpack RFC/changes.
